src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -2107,44 +1283,63 @@ private void CheckDisposedOrStarted()
                 }
             }
         }
-
-        private class RequestState
+        
+        private void ThrowOnInvalidHandle(SafeWinHttpHandle handle)
         {
-            public RequestState()
+            if (handle.IsInvalid)
             {
-                TransportContext = new WinHttpTransportContext();
+                throw new HttpRequestException(
+                    SR.net_http_client_execution_error,
+                    WinHttpException.CreateExceptionUsingLastError());
```

Comment:
In a review of CurlHandler, there was a suggestion to centralize the wrapping with HttpRequestException into a single place.  Should this just be throwing the CreateExceptionUsingLastError exception and having it wrappedin an HttpRequestException in some central location?